// graphql-playground-middleware/dist/hapi.js
"use strict";
var __assign = (this && this.__assign) || Object.assign || function(t) {
    for (var s, i = 1, n = arguments.length; i < n; i++) {
        s = arguments[i];
        for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
            t[p] = s[p];
    }
    return t;
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) if (e.indexOf(p[i]) < 0)
            t[p[i]] = s[p[i]];
    return t;
};
Object.defineProperty(exports, "__esModule", { value: true });
var render_playground_page_1 = require("./render-playground-page");
// tslint:disable-next-line
var pkg = require('../package.json');
// tslint:disable-next-line only-arrow-functions
var hapi = function (server, options, next) {
    if (arguments.length !== 3) {
        throw new Error("Playground middleware expects exactly 3 arguments, got " + arguments.length);
    }
    var path = options.path, _a = options.route, config = _a === void 0 ? {} : _a, voyagerOptions = __rest(options, ["path", "route"]);
    var middlewareOptions = __assign({}, voyagerOptions, { version: pkg.version });
    server.route({
        method: 'GET',
        path: path,
        config: config,
        handler: function (request, reply) {
            reply(render_playground_page_1.default(middlewareOptions)).header('Content-Type', 'text/html');
        },
    });
    return next();
};
hapi.attributes = {
    pkg: pkg,
    multiple: false,
};
exports.default = hapi;
